platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/events/lifecycle/AppExitLogger.kt

```diff
@@ -140,18 +140,38 @@ internal class AppExitLogger(
 
     @TargetApi(Build.VERSION_CODES.R)
     private fun ApplicationExitInfo.toFields(): InternalFieldsMap {
-        // https://developer.android.com/reference/kotlin/android/app/ApplicationExitInfo
-        return mapOf(
-            "_app_exit_source" to "ApplicationExitInfo",
-            "_app_exit_process_name" to this.processName,
-            "_app_exit_reason" to this.reason.toReasonText(),
-            "_app_exit_importance" to this.importance.toImportanceText(),
-            "_app_exit_status" to this.status.toString(),
-            "_app_exit_pss" to this.pss.toString(),
-            "_app_exit_rss" to this.rss.toString(),
-            "_app_exit_description" to this.description.orEmpty(),
-            // TODO(murki): Extract getTraceInputStream() for REASON_ANR or REASON_CRASH_NATIVE
-        ).toFields()
+        return buildMap {
+            put("_app_exit_source", "ApplicationExitInfo")
+            put("_app_exit_process_name", processName)
+            put("_app_exit_reason", reason.toReasonText())
+            put("_app_exit_importance", importance.toImportanceText())
+            put("_app_exit_status", status.toString())
+            put("_app_exit_pss", pss.toString())
+            put("_app_exit_rss", rss.toString())
+            put("_app_exit_description", description.orEmpty())
+
+            // Add trace data if available for REASON_ANR or REASON_CRASH_NATIVE
+            if (reason == ApplicationExitInfo.REASON_ANR || reason == ApplicationExitInfo.REASON_CRASH_NATIVE) {
+                runCatching {
+                    getTraceInputStream()?.takeIf { it.available() > 0 }?.bufferedReader()?.use { reader ->
```

Comment:
Another thing I noticed from [the docs](https://developer.android.com/reference/kotlin/android/app/ApplicationExitInfo#gettraceinputstream): 
>Note that because these traces are kept in a separate global circular buffer, crashes may be overwritten by newer crashes (including from other applications)

Something else that we might need to think about if and how to handle